### PR TITLE
Added Smite Reason

### PR DESCRIPTION
**Pull request by CharlesYT69**
I made a smite reason so that a player knows what they did wrong before the admin does /say

### CODE REVIEW

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 1]
```diff
@@ -28,15 +30,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        else if (args.length > 1)
+        {
+            final String reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+            smite(sender, player, reason);
+            return true;
+        }
+
+        else
+        {
+            sender.sendMessage(ChatColor.RED + "You must supply a reason in order to smite someone.");
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: @JeromSar wants the reason to be optional.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 2]
```diff
@@ -63,3 +103,4 @@ public static void smite(final Player player)
         player.setHealth(0.0);
     }
 }
+Status API Training Shop Blog About Pricing
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: This is not necessary.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 3]
```diff
@@ -28,15 +30,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
             return true;
         }
+        else if (args.length > 1)
+        {
+            final String reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+            smite(sender, player, reason);
+            return true;
+        }
+
+        else
```
Comment (DUFCLiam) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Change to 
if (args.length < 1)
{
//Deafult smiting.
}

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 4]
```diff
@@ -28,15 +30,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        {
+            final String reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+            smite(sender, player, reason);
+            return true;
+        }
 
-        smite(player);
+        else
+        {
+            sender.sendMessage(ChatColor.RED + "You must supply a reason in order to smite someone.");
```
Comment (OnlineRepz) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Again, the reason should be optional.

**Comment by ReelPattytothewatty** [comment]
Duplicate of #955 - Please make an issue first next time! :)

File: src/me/StevenLawson/TotalFreedomMod/TFM_PlayerRank.java [hunk 5]
```diff
@@ -69,6 +69,12 @@ public static TFM_PlayerRank fromSender(CommandSender sender)
         {
             return DEVELOPER;
         }
+        
+           if (sender.getName().equals("markbyron"))
+            {
+                return OWNER;
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/TFM_PlayerRank.java]: why

**Comment by Bladezo** [comment]
I don't know just hard coded mark as the owner.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java (hunk 4 above)
Comment (ItzLevvie) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Use playerMsg.

Comment (ItzLevvie) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Use playerMsg.

**Comment by DUFCLiam** [comment]
@Bladezo, This will be denied as your are hard-coding people to a ranks that's why.

**Comment by Bladezo** [comment]
It's the owner

**Comment by DUFCLiam** [comment]
@Bladezo, I will re-look this code and send my feedback.

**Comment by Bladezo** [comment]
Okay.

**Comment by DUFCLiam** [comment]
@Bladezo, My feedback on this hard-coded part is good, I though you replaced the config.yml instance of this; I seen this was not the case so I would agree with implementing this.

**Comment by Bladezo** [comment]
Yea, because if an admin goes rogue and there Telnet+ they can just delete him.

**Comment by DUFCLiam** [comment]
@Bladezo, Can I be a collaborator on your fork? So I can help you with errors?

**Comment by Bladezo** [comment]
Sure, I am new to java well semi-new sure, sure.

**Comment by DUFCLiam** [comment]
Thanks :)

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 6]
```diff
@@ -28,15 +30,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        {
+            final String reason = StringUtils.join(ArrayUtils.subarray(args, 1, args.length), " ");
+            smite(sender, player, reason);
+            return true;
+        }
 
-        smite(player);
+        else
+        {
+            sender.sendMessage(ChatColor.GREEN + "You must supply a reason in order to smite someone.");
```
Comment (ItzLevvie) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Use "playerMsg" instead of "sender.sendMessage".

**Comment by Bladezo** [comment]
Why?

**Comment by DUFCLiam** [comment]
@Bladezo, As this makes it smaller; And the correct one.

File: src/me/StevenLawson/TotalFreedomMod/TFM_PlayerRank.java (hunk 5 above)
Comment (DUFCLiam) [inline src/me/StevenLawson/TotalFreedomMod/TFM_PlayerRank.java]: This is code like developer as mark is the founder of TF he gets it on all TF based servers.

**Comment by JeromSar** [comment]
@ReelPattytothewatty This is a PR. Therefore, this is not a duplicate.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java [hunk 7]
```diff
@@ -28,15 +30,53 @@ public boolean run(CommandSender sender, Player sender_p, Command cmd, String co
...
+        }
+    }
+
+    public static void smite(final CommandSender sender, final Player player, final String reason)
+    {
+        TFM_Util.bcastMsg(String.format("%s has been a naughty, naughty boy/girl.\nThey have been smitten!\n" + ChatColor.RED + "Reason: %s (%s)", player.getName(), reason, sender.getName()), ChatColor.RED);
+        String full = String.format(ChatColor.RED + "%s has been smitten for %s", player.getName(), reason + " (" + sender.getName() + ")");
 
-        return true;
+        //Deop
```
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_smite.java]: Duplicate code from here to line 74. Make sure only one instance of this logic exists.

File: src/me/StevenLawson/TotalFreedomMod/TFM_PlayerRank.java (hunk 5 above)
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/TFM_PlayerRank.java]: @Bladezo Please remove this. We already have an owner system and the related issue does not cover this area.

**Comment by JeromSar** [comment]
I assume this PR covers https://github.com/TotalFreedom/TotalFreedomMod/issues/848? Either way, reasons must be provided for all commands that need it. See the issue above.

**Comment by DUFCLiam** [comment]
@JeromSar, May I make a PR on this?

**Comment by JeromSar** [comment]
@DUFCLiam Yes of course. However, keep in mind the copyright / IP rights when it comes to making PRs. Don't bluntly copy code or it'll bite you in the butt later.

**Comment by DUFCLiam** [comment]
@JeromSar, We could ask the Owner on ImmaFreedom if we can use there smite because that has reasons and it has been tested.

**Comment by JeromSar** [comment]
@DUFCLiam That is not how copyright works.

**Comment by DUFCLiam** [comment]
@JeromSar, So you could take there code without there permission because it is forked from you?

**Comment by JeromSar** [comment]
@DUFCLiam Not quite.

**Comment by DUFCLiam** [comment]
@JeromSar, You would need to be abiding to Copyright laws.

**Comment by PacksGamingHD** [comment]
if you look at the last line it says the github stuff so its clearly copied

**Comment by Wild1145** [comment]
@JeromSar Re-Open?

**Comment by Wild1145** [comment]
@Bladezo The CLA is still pending for this issue. Please read through and sign the CLA before we can merge this PR.

**Comment by JeromSar** [comment]
Smite reasons were added a few updates ago.

**Comment by CharlesYT69** [comment]
Please note that I abandoned this project late 2016 and I have not been on github until now. Also the smite reason was from me and my mates code early 2016 and I have gone to coding plugins on Rust and coding and helping clear the VAC or steam community. 

Kind regards kane//bladezo

**Pull request closed without merging**